.github/workflows/ci.yml

```diff
@@ -249,6 +249,34 @@ jobs:
         with:
           files: ./coverage.txt
 
+  vendor:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+          cache: true
+      -
+        name: Build vendor
+        run: |
+          go mod vendor
+          mkdir -p bin/release
+          tar czf bin/release/docker-compose-vendor.tar.gz vendor
+      -
+        name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: compose
```

Comment:
I see that the artifact always has the same name. The behaviour of this depends on the `overwrite` value. 

```
    # If true, an artifact with a matching name will be deleted before a new one is uploaded.
    # If false, the action will fail if an artifact for the given name already exists.
    # Does not fail if the artifact does not exist.
    # Optional. Default is 'false'
    overwrite:
```

This will have the default `overwrite` value of `false` so it will always fail after the first time, if I understand this correctly.